signingscript/src/signingscript/sign.py

```diff
@@ -811,27 +812,46 @@ def write_signing_req_to_disk(fp, signing_req):
     Does proper base64 and json encoding.
     Tries not to hold onto a lot of memory.
     """
-    fp.write(b"[{")
-    for k, v in signing_req.items():
-        fp.write(json.dumps(k).encode("utf8"))
-        fp.write(b":")
-        if hasattr(v, "read"):
-            # Make sure we're always reading from the beginning of the file
-            # Sometimes we have to retry the request
-            v.seek(0)
-            fp.write(b'"')
+    if "files" in signing_req:
+        # Don't mutate signing_req in case of retry
+        encoded_signing_req = signing_req.copy()
+        for input_file in encoded_signing_req["files"]:
+            encoded_file_content = StringIO()
+            input_stream = input_file["content"]
+            # Make sure we're always reading from the beginning of the file in case of retry
+            input_stream.seek(0)
             while True:
-                block = v.read(1020)
+                block = input_stream.read(1020)
                 if not block:
                     break
-                e = b64encode(block).encode("utf8")
-                fp.write(e)
-            fp.write(b'"')
-        else:
-            fp.write(json.dumps(v).encode("utf8"))
-        fp.write(b",")
-    fp.seek(-1, 1)
-    fp.write(b"}]")
+                encoded_block = b64encode(block)
+                encoded_file_content.write(encoded_block)
```

Comment:
I see a bit of potential for factoring out here, maybe just a `def _encode_single_file()` inner function that can be shared with below. But not a big deal to leave as is either.

Reply:
This cleaned up nicely thanks :)